### PR TITLE
mitigate epoch boundary attacks by randomizing cutoff

### PR DESCRIPTION
**Pull request by whyrusleeping**
using crypto rand because it needs to be very unguessable, and i didnt feel like seeding math/rand.

### CODE REVIEW

File: miner/miner.go [hunk 1]
```diff
@@ -46,7 +56,11 @@ func NewMiner(api api.FullNode, epp gen.WinningPoStProver, addr address.Address)
 		waitFunc: func(ctx context.Context, baseTime uint64) (func(bool, error), error) {
 			// Wait around for half the block time in case other parents come in
```
Comment (Kubuxu) [inline miner/miner.go:57]: This comment is a bit misleading, as `PropagationDelaySecs` is 6s.
Just saying.

File: miner/miner.go [hunk 2]
```diff
@@ -33,6 +35,14 @@ var log = logging.Logger("miner")
 // returns a callback reporting whether we mined a blocks in this round
 type waitFunc func(ctx context.Context, baseTime uint64) (func(bool, error), error)
 
+func randTimeOffset(width time.Duration) time.Duration {
+	buf := make([]byte, 8)
+	rand.Reader.Read(buf)
+	val := time.Duration(binary.BigEndian.Uint64(buf) % uint64(width))
+
+	return val - (width / 2)
```
Comment (Kubuxu) [inline miner/miner.go:43]: Why `width/2`, we always move the boundary back.

File: miner/miner.go (hunk 1 above)
Comment (whyrusleeping) [reply miner/miner.go:57]: oh yeah, thats been wrong for a while

File: miner/miner.go (hunk 2 above)
Comment (whyrusleeping) [reply miner/miner.go:43]: because I want it to be framed as 6s +/- 500ms

Comment (Kubuxu) [reply miner/miner.go:43]: ahh, I misread it :/

**Pull request merged**